### ISSUE TITLE
Integrate invenio-oauth2server

### ISSUE THREAD

**Issue by kaplun** [inspirehep/inspire-next#1008]

**Comment by jmartinm** [comment]
What are some of the access control use cases that we foresee?

Note that `invenio-records-rest` provides some hooks to have `read/update/create` permissions - https://github.com/inveniosoftware/invenio-records-rest/blob/master/invenio_records_rest/views.py#L76 so we can add the logic in our overlay.

**Comment by kaplun** [comment]
Well, the idea is that we could fully protect /api behind oauth authorization so that we can carefully control who access it. I wonder if also the flask ratelimit support can be bound to an oauth token so that we can limit the amount of query a given client is going to perform on our system.

**Comment by StellaCh** [comment]
Is this still relevant @kaplun ?

**Comment by kaplun** [comment]
It's relevant if you will setup collaboration with partners where data is *pushed* to INSPIRE, or if you want to have some control and monitor who instead is *getting* data from INSPIRE.

It's more a strategic decision to be taken. It's not urgent for the time being.

**Comment by salmele** [comment]
Different level of access control are to be expected for pushing into INSPIRE (think the controls we would have to create curator accounts) or to pull data from INSPIRE. The latter might be fully open (modulo credentials to ensure proper use wrt bandwidth and architecture of nodes)

**Comment by kaplun** [comment]
Yup. Access control for regular user web access is already in place. OAUTH would be there to allow machine access of INSPIRE in an ad-hoc way.
